kafka-protocol-binding.md

```diff
@@ -130,14 +142,25 @@ this specification then it is not a valid CloudEvent.
 If the `content-type` header is not present then the receiver uses
 _structured_ mode with the JSON event format.
 
-### 3.1. Key Attribute
+### 3.1. Key Mapping
+
+The 'key' of the Kafka message is populated by a "Key Mapper" function, which
+might map the key directly from one of the CloudEvent's attributes, but might
+also use information from the application environment, from the CloudEvent's
+data or other sources.  
+
+The shape and configuration of the "Key Mapper" function is implementation
+specific.
+
+Every implementation SHOULD provide a default "Key Mapper" implementation that
+maps the [Partitioning](extensions/partitioning.md) `partitionkey` attribute
+value to the 'key' of the Kafka message as-is, if present. 
 
-The 'key' attribute is populated by a partitionKeyExtractor function. The
-partitionKeyExtractor is a protocol specific function that contains bespoke
-logic to extract and populate the value. The key attribute MUST be encoded
-as UTF-8 string both in the Kafka message and in the CloudEvents extension.
-A default implementation of the extractor will use the
-[Partitioning](extensions/partitioning.md) extension value.
+A mapping function MUST NOT modify the CloudEvent. This means that the
+aforementioned `partitionkey` attribute MUST still be included with the
+transmitted event. It also means that a mapping function that uses key
```

Comment:
> transmitted event.

transmitted event, if present.

Reply:
That condition is still in the paragraph above, no?

Reply:
It doesn't hurt to repeat IMO, @duglin?